internal/controller/managedcluster_controller.go

```diff
@@ -270,55 +275,162 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, l logr.Logger, ma
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
-			} else {
-				return ctrl.Result{}, err
-			}
+	if !fluxconditions.IsReady(hr) {
+		return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
+	}
+
+	return r.updateServices(ctx, l, managedCluster)
+}
+
+// updateServices reconciles services provided in ManagedCluster.Spec.Services.
+// TODO(https://github.com/Mirantis/hmc/issues/361): Set status to ManagedCluster object at appropriate places.
+func (r *ManagedClusterReconciler) updateServices(ctx context.Context, l logr.Logger, mc *hmc.ManagedCluster) (ctrl.Result, error) {
+	opts := []sveltos.HelmChartOpts{}
+
+	// NOTE: The ClusterProfile object will be updated with no helm
+	// charts if len(mc.Spec.Services) == 0. This will result in the
+	// helm charts being uninstalled on matching clusters if
+	// ClusterProfile originally had len(m.Spec.Sevices) > 0.
+	for _, svc := range mc.Spec.Services {
+		if !svc.Install {
+			l.Info(fmt.Sprintf("Skip adding Template (%s) to ClusterProfile (%s) because install=false", svc.Template, sveltos.ClusterProfileName(mc.Namespace, mc.Name)))
+			continue
 		}
 
-		if requeue {
-			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+		tmpl := &hmc.ServiceTemplate{}
+		tmplRef := types.NamespacedName{Name: svc.Template, Namespace: mc.Namespace}
+		if err := r.Get(ctx, tmplRef, tmpl); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get Template (%s)", tmplRef.String())
 		}
 
-		if !fluxconditions.IsReady(hr) {
-			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+		url, err := r.getServiceTemplateRepoURL(ctx, tmpl)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not get repository url: %w", err)
 		}
+
+		opts = append(opts, sveltos.HelmChartOpts{
+			RepositoryURL:  url,
+			RepositoryName: tmpl.Spec.Helm.ChartName,
+			ChartName:      tmpl.Spec.Helm.ChartName,
+			ChartVersion:   tmpl.Spec.Helm.ChartVersion,
+			ReleaseName:    svc.ReleaseName,
+			Values:         svc.Values,
+			ReleaseNamespace: func() string {
+				if svc.ReleaseNamespace != "" {
+					return svc.ReleaseNamespace
+				}
+				return svc.ReleaseName
+			}(),
+			CreateNamespace: svc.CreateNamespace,
+			PlainHTTP:       svc.RegistryConfig.PlainHTTP,
+			Insecure:        svc.RegistryConfig.Insecure,
+		})
+	}
+
+	cp, _, err := sveltos.ReconcileClusterProfile(ctx, r.Client, mc.Namespace, mc.Name,
+		map[string]string{
+			hmc.FluxHelmChartNamespaceKey: mc.Namespace,
+			hmc.FluxHelmChartNameKey:      mc.Name,
+		},
+		sveltos.ReconcileClusterProfileOpts{
+			OwnerReference: &metav1.OwnerReference{
+				APIVersion: hmc.GroupVersion.String(),
+				Kind:       hmc.ManagedClusterKind,
+				Name:       mc.Name,
+				UID:        mc.UID,
+			},
+			HelmChartOpts: opts,
+		})
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create ClusterProfile: %w", err)
 	}
-	return ctrl.Result{}, nil
+
+	l.Info(fmt.Sprintf("Successfully created ClusterProfile (%s)", cp.Name))
```

Comment:
This log will be shown every time the ClusterProfile is created/updated or not changed. I'd move it to the `ReconcileClusterProfile` function instead and handle the logging message depending on the `operation` result.